src/adapter/vertex.rs

```diff
@@ -36,6 +36,7 @@ pub enum VertexKind<'a> {
     ImplementedTrait(&'a Path, &'a Item),
     FunctionParameter(&'a str),
     FunctionAbi(&'a Abi),
+    Discriminant(Discriminant),
```

Comment:
Any reason not to return `&'a Discriminant` and avoid cloning?
```suggestion
    Discriminant(&'a Discriminant),
```

Reply:
This is definitely better; was just rushing to get something working before the end of the hackathon and so didn't want to deal with lifetimes.